### PR TITLE
[SYCL-MLIR] Make accessor raising more robust

### PR DESCRIPTION
**Pull request by sommerlukas**
Make the raising for buffers and accessors more robust by allowing number suffixes for the allocated type name and matching with a regex.

### CODE REVIEW

**Review by victor-eds — COMMENTED** [review]
Can you also add a `CHECK` line in

**Review by victor-eds — COMMENTED** [review]
Can you also add a `CHECK` line [here](https://github.com/intel/llvm/blob/8f01166a32265a8b6185d5bafd17cfea19377f4c/polygeist/tools/cgeist/Test/Verification/sycl/host_raising.cpp#L26)?

File: polygeist/lib/Dialect/Polygeist/Transforms/SYCLRaiseHost.cpp [hunk 1]
```diff
@@ -343,7 +353,10 @@ struct BufferInvokeConstructorPattern
 
 struct AccessorTypeTag {
 
-  static llvm::StringRef getTypeName() { return "class.sycl::_V1::accessor"; }
+  static llvm::Regex &getTypeName() {
+    static llvm::Regex regex{"class.sycl::_V1::accessor(\\.[0-9]+])?"};
```
[the remark this answers is not in the document]
Comment (sommerlukas) [reply polygeist/lib/Dialect/Polygeist/Transforms/SYCLRaiseHost.cpp:357]: What would be the advantage from your point of view?

**Review by jopperm — APPROVED** [review]

**Review by victor-eds — APPROVED** [review]

File: polygeist/tools/cgeist/Test/Verification/sycl/host_raising.cpp [hunk 2]
```diff
@@ -23,6 +23,7 @@ class KernelName;
 // CHECK-LABEL: llvm.func internal @_ZNSt17_Function_handlerIFvRN4sycl3_V17handlerEEZ4mainEUlS3_E_E9_M_invokeERKSt9_Any_dataS3_
 // CHECK:          sycl.host.constructor({{.*}}) {type = !sycl_accessor_1_21llvm2Evoid_r_gb} : (!llvm.ptr, !llvm.ptr, !llvm.ptr, !llvm.ptr, !llvm.ptr) -> ()
 // CHECK:          sycl.host.constructor({{.*}}) {type = !sycl_accessor_1_21llvm2Evoid_r_gb} : (!llvm.ptr, !llvm.ptr, !llvm.ptr, !llvm.ptr, !llvm.ptr) -> ()
+// CHECK:          sycl.host.constructor({{.*}}) {type = !sycl_accessor_1_21llvm2Evoid_w_gb} : (!llvm.ptr, !llvm.ptr, !llvm.ptr, !llvm.ptr, !llvm.ptr) -> ()
```
Comment (victor-eds) [inline polygeist/tools/cgeist/Test/Verification/sycl/host_raising.cpp:26]: Nice

**Pull request merged**